pkg/controller/propagator/propagation.go

```diff
@@ -676,22 +685,37 @@ func (r *ReconcilePolicy) processTemplates(replicatedPlc *policiesv1.Policy, dec
 			reqLogger.Error(tplErr, "Failed to resolve templates")
 
 			r.recorder.Event(rootPlc, "Warning", "PolicyPropagation",
-				fmt.Sprintf("Failed to resolve templates for policy %s/%s for cluster %s/%s .", rootPlc.GetName(), rootPlc.GetNamespace(), decision.ClusterNamespace, decision.ClusterName))
+				fmt.Sprintf("Failed to resolve templates for cluster %s/%s: %s", decision.ClusterNamespace, decision.ClusterName, tplErr.Error()))
+
+			//Set an annotation on the policyTemplate(e.g. ConfigurationPolicy)  to the template processing error msg
+			//managed clusters will use this when creating a violation
+			policyTObjectUnstructured := &unstructured.Unstructured{}
+			jsonErr := json.Unmarshal(policyT.ObjectDefinition.Raw, policyTObjectUnstructured)
+			if jsonErr != nil {
+				//it shouldnt get here but if it did just log a msg
+				//its alright, a generic msg will be used on the managedcluster
+				reqLogger.Error(jsonErr, "Error unmarshalling to json")
```

Comment:
Could you please add more context to the log message such as which policy this is referring to?

Reply:
fixed